generators/spring-data-relational/templates/src/test/java/package/config/timezone/HibernateTimeZoneIT.java.ejs

```diff
@@ -54,6 +56,12 @@ class HibernateTimeZoneIT {
     private DateTimeFormatter timeFormatter;
     private DateTimeFormatter dateFormatter;
 
+    @BeforeAll
+    static void beforeAll() {
+        // Run tests in non-UTC timezone but avoid possible issues due to DTS
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC+1"));
```

Comment:
As far as I understand, this test ensures that everything is stored in the UTC time zone, even if the application is run in a non-UTC time zone. At the same time, I was not able to get a consistent result in a time zone with DST, perhaps JDBC drivers handle such cases differently.
At the same time, to be honest, I don’t see much point in “testing the framework” at all.

Reply:
Yeah, I also was trying to understand what we are trying to test here.

It seems that we are trying to assure that dates are being stored in the UTC timezone in DB, and no matter what time zone was there in the raw input (at least the `assertThatDateStoredValueIsEqualToInsertDateValueOnGMTTimeZone()` method looks exactly like that).
I suppose it's the behaviour we're trying to achieve in production.

But let's look at what we have in `src/main/resources/config/application-prod.yml` for, let's say, MySQL. And there we can see the following param in the URL: `&serverTimezone=UTC`. AFAIK, it's there because otherwise, the DB would be storing dates in its default (system) time zone instead of UTC in any case.

So then, the reason why tests are failing is that the H2 db stores dates in its system time zone because it's not overridden (`TIME ZONE=UTC` isn't present in our configurations).
I was also looking into the TestContainers problem and found that `SqlTestContainersSpringContextCustomizerFactory` has no `&serverTimezone=UTC` param, the one we have there for production. So dates are being stored in the system time zone instead of UTC (except in the case the devs machine is in UTC).

I believe we either should be setting the time zone for all databases or none of them.

I agree that tests should pass in any time zone but I believe "in any time zone" means the time zone of the application, but the time zone of the DB still should be UTC.
Otherwise, we shouldn't assert on Strings but on real date objects.

Does it make any sense?

Reply:
@shrralis  when I create a project with `"devDatabaseType": "mysql"` I see that `SqlTestContainersSpringContextCustomizerFactory` has `&serverTimezone=UTC` but even setting `TIME ZONE=UTC` in H2 database, unfortunately, doesn't make database records the same.

H2 database with `TIME ZONE=UTC`:
```
2023-08-09T12:51:38.797+02:00 TRACE 3668 --- [           main] o.h.r.j.i.ResourceRegistryStandardImpl   : Releasing statement [HikariProxyPreparedStatement@523455712 wrapping prep2: insert into jhi_date_time_wrapper (instant,local_date,local_date_time,local_time,offset_date_time,offset_time,zoned_date_time,id) values (?,?,?,?,?,?,?,?) {1: TIMESTAMP '2014-11-12 05:50:00', 2: DATE '2016-09-09', 3: TIMESTAMP '2014-11-12 06:50:00', 4: TIME '13:30:00', 5: TIMESTAMP WITH TIME ZONE '2011-12-14 08:30:00+00', 6: TIME WITH TIME ZONE '12:30:00+00', 7: TIMESTAMP WITH TIME ZONE '2011-12-14 08:30:00+00', 8: CAST(20851 AS BIGINT)}]
```

MySQL with `serverTimezone=UTC`:
```
2023-08-09T12:51:56.187+02:00 TRACE 3777 --- [           main] o.h.r.j.i.ResourceRegistryStandardImpl   : Releasing statement [HikariProxyPreparedStatement@19442926 wrapping com.mysql.cj.jdbc.ServerPreparedStatement[5]: insert into jhi_date_time_wrapper (instant,local_date,local_date_time,local_time,offset_date_time,offset_time,zoned_date_time) values ('2014-11-12 05:50:00','2016-09-10','2014-11-12 06:50:00','13:30:00','2011-12-14 08:30:00','11:30:00','2011-12-14 08:30:00')]
```

I also played around with the various settings and their combinations described [here](https://dev.mysql.com/doc/connector-j/8.1/en/connector-j-time-instants.html), but nothing worked to get consistent results without forcing the timezone in the test.

Reply:
Unfortunately we are having fundamental behavior changes across databases using `NORMALIZE_UTC`.

Reply:
The reason for `NORMALIZE` was that `NORMALIZE_UTC` results are not consistent (h2 to blame, tests are passing at others databases).
So it's not a viable candidate to be used by default right now.

Reply:
@mshima another thing we could try is to switch from H2 to HSQLDB for an in-memory database.

On the other hand, I still think that maybe it's even better that we're staying in backwards compatible mode for now rather than trying to stick to the default. This feature in Hibernate is still marked as `@Incubating` and the default may change again in future releases.

Anyway, I'll add some JavaDoc tonight or tomorrow so we could close the issue.

Reply:
I think it's more likely for us to drop h2 official support than switch to hsqldb.
Using Docker is recommended to have same behavior at dev and prod.

Reply:
@mshima I've rebuild project with latest changes but `storeOffsetTimeWithZoneIdConfigShouldBeStoredNormalizedToSystemTimeZone` still fails for me (I'm in Europe/Stockholm timezone), so I will also fix it besides adding the Javadoc.